src/InvariantTheory/invariant_rings.jl

```diff
@@ -127,16 +127,81 @@ function Base.show(io::IO, IR::InvRing)
 end
 
 function reynolds_molien_via_singular(IR::InvRing)
+   @assert !ismodular(IR)
    if !isdefined(IR, :reynolds_singular) || !isdefined(IR, :molien_singular)
       R = polynomial_ring(IR)
       singular_matrices = _action_singular(IR)
-      rey, mol = Singular.LibFinvar.reynolds_molien(singular_matrices...)
+
+      if iszero(characteristic(coefficient_ring(IR)))
+         rey, mol = Singular.LibFinvar.reynolds_molien(singular_matrices...)
+      else
+         error("Not implemented yet") # See https://github.com/oscar-system/Singular.jl/issues/486
+      end
       IR.reynolds_singular = rey
       IR.molien_singular = mol
    end
    return IR.reynolds_singular, IR.molien_singular
 end
 
+function reynolds_operator(IR::InvRing{FldT, GrpT, T}, f::T) where {FldT, GrpT, T <: MPolyElem}
```

Comment:
You moved `PolyElemT` from last position to 3rd; but why not to first, then you could simplify the above to `reynolds_operator(IR::InvRing{T}, f::T) where {T <: MPolyElem}`, couldn't you?